pmd-java/etc/grammar/Java.jjt

```diff
@@ -2510,7 +2854,7 @@ void SwitchLabel() :
 {
 { inSwitchLabel = true; }
 (
-  "case" ( ConditionalExpression() #Expression ) ({checkForMultipleCaseLabels();} "," ( ConditionalExpression() #Expression ) )*
+  "case" (ConditionalExpression()) ({checkForMultipleCaseLabels();} "," (ConditionalExpression()) )*
```

Comment:
Expression is not a node anymore so it should be expected that you can find no ASTExpression node here, only a subtype. 

ConditionalExpression can expand to VariableReference (through PrimaryExpression) so it allows EnumConstantName (a single identifier). The node that is pushed is not a Name, it's a VariableReference, and the job of finding out whether this is an enum constant name or another static constant's name is left to the symbol table.

The JLS writes it this way because ConstantExpression may only evaluate to a primitive type or a string, so it needs a provision in this specific case to include EnumConstant. But a parser doesn't need to care, the determination of whether an expression is constant or not is done after a symbol table has been built anyway. This is yet another case where the JLS puts semantics in its BNF